overlord/hookstate/ctlcmd/set.go

```diff
@@ -165,3 +165,46 @@ func (s *setCommand) setInterfaceSetting(context *hookstate.Context, plugOrSlot
 	attrsTask.Set(which, attributes)
 	return nil
 }
+
+func copyAttributes(value map[string]interface{}) (map[string]interface{}, error) {
+	cpy, err := copyRecursive(value)
+	if err != nil {
+		return nil, err
+	}
+	return cpy.(map[string]interface{}), err
+}
+
+func copyRecursive(value interface{}) (interface{}, error) {
+	switch v := value.(type) {
+	case string:
+		return v, nil
+	case bool:
+		return v, nil
+	case int:
```

Comment:
this case should not exist because validateAttr normalizes  int to int64 , we need to do the same in code that deals with stuff set by snapctl btw